Add --pacman-cache, --pacman-log and --maxdepth to CLI options

atreyasha opened the pull request:
This PR addresses #120 by adding `--pacman-cache`, `--pacman-log` and `--maxdepth` as CLI options. Commits have been chunked to represent discrete components of this PR.

Regarding whether to include `--maxdepth` at all, I think it is a good idea to include this parameter to ensure that `find` does not get stuck searching some deep user-defined cache directory. The user can add an arbitrarily large value for this if they want to search some basic cache directory for (eg.) AUR packages.

pbrisbin commented on downgrade:
```diff
@@ -159,7 +169,7 @@ search_packages() {
 
     # shellcheck disable=SC2086
-    find $PACMAN_CACHE -maxdepth 1 -regextype posix-extended -regex ".*/$pkgfile_re"
+    find $PACMAN_CACHE -maxdepth $MAXDEPTH -regextype posix-extended -regex ".*/$pkgfile_re"
```
`shellcheck disable=SC2086` is here because we _want_ `$PACMAN_CACHE` to word-split if necessary, but `$MAXDEPTH` can (and should) be quoted.

atreyasha replied:
I see, will do.

pbrisbin commented on downgrade:
```diff
@@ -299,6 +309,7 @@ main() {
 : "${DOWNGRADE_FROM_ALA:=1}"
 : "${DOWNGRADE_FROM_CACHE:=1}"
 : "${DOWNGRADE_NOSUDO:=0}"
+: "${MAXDEPTH:=1}"
```
Please use `DOWNGRADE_MAXDEPTH`, to be consistent with the others. I don't like tools that expect un-namespaced environment variables, as there can be collisions or just confusion.

atreyasha replied:
Done.

pbrisbin commented on doc/downgrade.8.md:
```diff
@@ -92,18 +106,14 @@ As per the usage syntax, any options supplied after the **\--** character sequen
 ## DEFAULT BEHAVIORS
 
-By default, **downgrade** will search both local caches and the ALA.
+By default, **downgrade** will search both local caches and the ALA. 
```
:art: you introduced some trailing whitespace here.

atreyasha replied:
Woops, fixed.

pbrisbin commented on doc/downgrade.8.md:
```diff
@@ -60,20 +60,34 @@ The columns have the following meaning:
 # OPTIONS
 
-## DOWNGRADE OPTIONS
+## DOWNGRADE OPTIONS WITH ARGUMENTS
```
FWIW, I don't it particularly valuable to have distinct sections.

atreyasha replied:
Will revert back.

pbrisbin commented on completion/zsh:
```diff
@@ -4,6 +4,9 @@ _downgrade(){
              '--pacman-conf[Specify pacman configuration file]:config:_files' \
+             '--pacman-cache[Specify pacman cache location(s)]:config:_files' \
+             '--pacman-log[Specify pacman log-file location]:config:_files' \
+             '--maxdepth[Specify maximum depth for local cache find]' \
```
Do you have to do something here to indicate it takes an argument?

atreyasha replied:
As far as I know, there is no autocomplete prompt for integers here. We can prompt it to complete files or paths as per the options above, but that is not the case for `--maxdepth`.

pbrisbin replied:
Well, I wasn't expecting it to be smart about completing integers, I was just afraid that if you did

```
% downgrade --maxdepth <tab>
```

and it doesn't know that `--maxdepth` take _some_ argument, it'll start completing more `downgrade` options at that position, instead of stopping like actual `find` does:

![](https://files.pbrisbin.com/screenshots/screenshot.548719.png)

atreyasha replied:
Ah I understand now! That should not be hard to do. Let me look through something and update the PR.

atreyasha replied:
Fixed, now it will not complete any option for `--maxdepth` and `--pacman`. I modified such that it will complete directories and not files for `--pacman-cache`.

atreyasha commented:
Made the respective fixes. Removed the commits for the man-doc to prevent conflicts. I had to keep the changes to the locale since I added translations for new keys. If all is fine, you can merge it.

atreyasha commented:
Is it okay if I do the merging to master and fixing of the conflicts directly in master?

pbrisbin commented:
> I do the merging to master (for both PRs) and fixing of the conflicts directly in master?

You should be able to,

- Merge one of them (GitHub UI shows no conflicts)
- Rebase the other one, fix conflicts
- Merge the other one

I don't think anything should need fixing "in master", right?

atreyasha commented:
Ah sorry I worded that wrongly, there should be no conflicts in `master` itself. There will probably be some conflicts before the merge but I can fix those in the UI directly. Will do it then.

atreyasha commented:
Merging done for both. Just need to rebuild the man-page/changelog which I will leave to you.

pbrisbin commented:
And just double-checking, major version bump because we changed how `=` works?

atreyasha commented:
Yes major version bump sounds right.

pbrisbin commented:
:tada: 8.0.0 released
